src/pydocstyle/conventions.py

```diff
@@ -0,0 +1,107 @@
+"""This module contains the convention definitions."""
+
+from typing import Set
+
+from typing_extensions import Literal
+
+from pydocstyle.violations import all_errors
+
+CONVENTION_NAMES = ("pep257", "numpy", "google")
+
+
+convention_errors = {
+    'pep257': all_errors
+    - {
+        'D203',
+        'D212',
+        'D213',
+        'D214',
+        'D215',
+        'D404',
+        'D405',
+        'D406',
+        'D407',
+        'D408',
+        'D409',
+        'D410',
+        'D411',
+        'D413',
+        'D415',
+        'D416',
+        'D417',
+        'D418',
+    },
+    'numpy': all_errors
+    - {
+        'D107',
+        'D203',
+        'D212',
+        'D213',
+        'D402',
+        'D413',
+        'D415',
+        'D416',
+        'D417',
+    },
+    'google': all_errors
+    - {
+        'D203',
+        'D204',
+        'D213',
+        'D215',
+        'D400',
+        'D401',
+        'D404',
+        'D406',
+        'D407',
+        'D408',
+        'D409',
+        'D413',
+    },
+}
+
+
+class Convention:
+    """This class defines the convention to use for checking docstrings."""
+
+    def __init__(
+        self, name: Literal["pep257", "numpy", "google"] = "pep257"
```

Comment:
We should prob just type this as a `str` instead of a `Literal`, then we don't need typing_extensions. The point of [PEP 586](https://peps.python.org/pep-0586/) (which introduced `Literal`) was to support functions that change return type based on which literal value is passed as input. Since we don't need that here, the type checker doesn't really benefit from using Literal over str.

Reply:
I like using `Literal` when the number of options is limited because static analysis will save me when I make a typo. However, I like your proposal of raising an error at runtime instead :+1: